extensions/cocostudio/timeline/CCBoneNode.js

```diff
@@ -0,0 +1,573 @@
+/****************************************************************************
+ Copyright (c) 2015-2016 Chukong Technologies Inc.
+
+ http://www.cocos2d-x.org
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in
+ all copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+ ****************************************************************************/
+
+/**
+ * BoneNode
+ * base class
+ * @class
+ */
+ccs.BoneNode = (function () {
+
+    var Node = cc.Node;
+    var SkinNode = ccs.SkinNode;
+    var BlendFunc = cc.BlendFunc;
+    var type = {
+        p: cc.p,
+        size: cc.size,
+        rect: cc.rect
+    };
+    var debug = {
+        log: cc.log,
+        assert: cc.assert
+    };
+
+    var BoneNode = Node.extend(/** @lends ccs.BoneNode# */{
+        _customCommand: null,
+        _blendFunc: null,
+
+        _rackColor: null,
+
+        _rackLength: null,
+        _rackWidth: null,
+
+        _childBones: null,
+        _boneSkins: null,
+        _rootSkeleton: null,
+
+        _squareVertices: null,
+        _squareColors: null,
+        _noMVPVertices: null,
+
+        ctor: function (length) {
+            // null
+            // length
+            Node.prototype.ctor.call(this);
+            // _isRackShow -> _renderCmd._debug
+            if (this._squareVertices === null)
+                this._squareVertices = [
+                    {x: 0, y: 0}, {x: 0, y: 0}, {x: 0, y: 0}, {x: 0, y: 0}
+                ];
+
+            this._rackColor = cc.color.WHITE;
+            this._rootSkeleton = null;
+            this._blendFunc = BlendFunc.ALPHA_NON_PREMULTIPLIED;
+
+            this._childBones = [];
+            this._boneSkins = [];
+
+            this._anchorPoint = type.p(0, .5);
```

Comment:
the latest code remove set archorpoint and contentsize's value
https://github.com/cocos2d/cocos2d-x/pull/12973